internal/venice-client-common/src/main/java/com/linkedin/venice/compression/ZstdWithDictCompressor.java

```diff
@@ -67,7 +67,7 @@ public InputStream decompress(InputStream inputStream) throws IOException {
 
   @Override
   public void close() throws IOException {
-    compressor.close();
+    compressor.get().close();
```

Comment:
Actually, there is a problem here... this will result in closing only the compressor associated with the thread calling close, but none of the others... this is a leak. Using `ThreadLocal` introduces some trickiness. We may need this class to hold a collection of compressors, which get added to by the `ThreadLocal` initializing closure... and then in this function iterate over all of them and close them all. May be good to revise our other usages of TL to figure out if we have other leaks elsewhere as well...

Reply:
Yes, I realized this last night and am adding a `CloseableThreadLocal` class